test/upgrade/prober/sut/default_e2e_test.go

```diff
@@ -0,0 +1,160 @@
+// +build e2e
+
+/*
+Copyright 2021 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package sut_test
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+	"testing"
+
+	cloudevents "github.com/cloudevents/sdk-go/v2"
+	"github.com/stretchr/testify/assert"
+	"go.uber.org/zap"
+	"k8s.io/apiserver/pkg/storage/names"
+	testlib "knative.dev/eventing/test/lib"
+	"knative.dev/eventing/test/lib/recordevents"
+	"knative.dev/eventing/test/lib/resources"
+	"knative.dev/eventing/test/upgrade/prober/sut"
+	watholaevent "knative.dev/eventing/test/upgrade/prober/wathola/event"
+	watholasender "knative.dev/eventing/test/upgrade/prober/wathola/sender"
+	"knative.dev/pkg/apis"
+	duckv1 "knative.dev/pkg/apis/duck/v1"
+	"knative.dev/pkg/signals"
+	pkgTest "knative.dev/pkg/test"
+)
+
+func TestSUTNewDefaultE2E(t *testing.T) {
```

Comment:
It's good you had this test when implementing this but I'm not sure we need to have it in the repo and run it continuously. Especially not as an E2E test for the upgrade test framework itself which is run as part of general E2E test suite (in the tests-e2e script). I'd just suggest removing it from this repo. Or convert it to a couple of unit tests which doesn't need the cluster to run.

Reply:
First, I wanted to have this as unit test. The one that uses fake K8s clients to simulate K8s cluster, but that turned out to be impossible, as there's unfortunately no fake client for `testlib.Client` that is ready to use. This test is executed - see: 

https://github.com/cardil/eventing/blob/5a7dcef33c37fc670c1cda47cea4473ec33afa79/test/e2e-tests.sh#L42-L43

BTW, I think, developing such fake client would be extremely beneficial, as we could use it to quickly run most of our e2e tests for quick check before using real cluster

Reply:
OK. Anyway, some tests are good at the beginning when developing the feature but don't need to be executed continuously. And I think this test falls into that group. BTW, the upgrade tests in this repo will continuously use this new implementation and they already run in CI. So, no need to run another test using the same bits. This is already tested enough, IMO.

Reply:
Yeah, that's also true. But, I think that it might be beneficial to leave this test for now. We could use it to easily implement & test other suts for: SinkBinding, Channel+Subscription.

Reply:
Is it possible to just have the test but do NOT execute it on every E2E test run? This make the test execution longer for every single build but I don't see the benefit of running it all the time.

Reply:
Okay. That makes sense. 

I'll add an appropriate comment on that test and remove the execution.